### PR TITLE
perf(apikey): add schema and feature flag for api key

### PR DESCRIPTION
**Pull request by KennethBWSong**

### CODE REVIEW

File: packages/fx-core/src/common/constants.ts [hunk 1]
```diff
@@ -59,4 +59,5 @@ export class FeatureFlagName {
   static readonly ApiCopilotPlugin = "API_COPILOT_PLUGIN";
   static readonly SampleConfigBranch = "TEAMSFX_SAMPLE_CONFIG_BRANCH";
   static readonly TestTool = "TEAMSFX_TEST_TOOL";
+  static readonly ApiKey = "TEAMSFX_API_KEY";
```
Comment (SLdragon) [inline packages/fx-core/src/common/constants.ts:62]: I think we can use a more specific name such as API_COPILOT_API_KEY

**Comment by SLdragon** [comment]
Also ask Aocheng to take a look about the schema changes

**Comment by KennethBWSong** [comment]
> Also ask Aocheng to take a look about the schema changes

@a1exwang Please help to review this change and see whether this pr will have any influence on your changes. Thanks!

**Review by a1exwang — APPROVED** [review]
LGTM for the schema part

**Review by jayzhang — APPROVED** [review]

**Review by 1yefuwang1 — APPROVED** [review]

**Pull request merged**